src/ui/shared/requests/uiInterpretTransaction.ts

```diff
@@ -0,0 +1,131 @@
+import { hashQueryKey, useQuery } from '@tanstack/react-query';
+import {
+  interpretSignature,
+  interpretTransaction,
+} from 'src/modules/ethereum/transactions/interpret';
+import type { IncomingTransactionWithChainId } from 'src/modules/ethereum/types/IncomingTransaction';
+import { getNetworksStore } from 'src/modules/networks/networks-store.client';
+import { invariant } from 'src/shared/invariant';
+import { normalizeChainId } from 'src/shared/normalizeChainId';
+import {
+  getPreferences,
+  usePreferences,
+} from 'src/ui/features/preferences/usePreferences';
+import { fetchAndAssignPaymaster } from 'src/modules/ethereum/account-abstraction/fetchAndAssignPaymaster';
+import { Client } from 'defi-sdk';
+import { ZerionAPI } from 'src/modules/zerion-api/zerion-api.client';
+import { useDefiSdkClient } from 'src/modules/defi-sdk/useDefiSdkClient';
+import { useCurrency } from 'src/modules/currency/useCurrency';
+import type { EligibilityQuery } from 'src/ui/components/address-action/EligibilityQuery';
+import { walletPort } from '../channels';
+
+/**
+ * Interprets Paymaster Transaction if it is eligible
+ * or a regular transaction otherwise
+ */
+export async function interpretTxBasedOnEligibility({
+  transaction,
+  eligibilityQueryData,
+  eligibilityQueryStatus,
+  currency,
+  origin,
+  client,
+}: {
+  transaction: IncomingTransactionWithChainId;
+  eligibilityQueryData: boolean | undefined;
+  eligibilityQueryStatus: 'error' | 'success' | 'loading';
+  currency: string;
+  origin: string;
+  client: Client;
+}) {
+  const preferences = await getPreferences();
+  const source = preferences?.testnetMode?.on ? 'testnet' : 'mainnet';
+  const networksStore = await getNetworksStore();
+  const networks = await networksStore.loadNetworksByChainId(
+    normalizeChainId(transaction.chainId)
+  );
+  const chain = networks.getChainById(normalizeChainId(transaction.chainId));
+  const network = chain ? networks?.getNetworkByName(chain) || null : null;
+  invariant(network, `Unidentified network: ${transaction.chainId}`);
+  if (!network.supports_simulations) {
```

Comment:
should we return null while `eligibilityQueryStatus === 'loading'`?

Reply:
Hm, let me check

Reply:
fixed